components/PralineForm/index.js

```diff
@@ -0,0 +1,216 @@
+import React, { useState } from "react";
+import { uid } from "uid";
+import useSWR from "swr";
+import { CldUploadButton } from "next-cloudinary";
+import Image from "next/image";
+import IngredientList from "./IngredientList";
+import InputField from "./InputField";
+
+export default function ProductForm() {
+  const [zutaten, setZutaten] = useState([]);
+  const [allergene, setAllergene] = useState([]);
+  const [currentPraline, setCurrentPraline] = useState(null);
+  const [nameField, setNameField] = useState("");
+  const [versionField, setVersionField] = useState("");
+  const [weightField, setWeightField] = useState("");
+  const [imageId, setImageId] = useState(null);
+
+  function handleAddIngredient(newIngredient) {
+    setZutaten([...zutaten, { ...newIngredient, id: uid() }]);
+  }
+
+  function handleDeleteIngredient(id) {
+    setZutaten(zutaten.filter((zutat) => zutat.id !== id));
+  }
+
+  function handleAddAllergen(allergeneMitMenge) {
+    setAllergene([...allergene, { ...allergeneMitMenge, id: uid() }]);
+  }
+
+  function handleDeleteAllergen(id) {
+    setAllergene(allergene.filter((allergen) => allergen.id !== id));
+  }
+
+  function cancel() {
+    setZutaten([]);
+    setAllergene([]);
+    setCurrentPraline(null);
+    setNameField("");
+    setVersionField("");
+    setWeightField("");
+  }
+
+  const { data, isLoading, mutate } = useSWR("/api/pralinen");
+  // console.log("data", data);
+
+  if (isLoading) {
+    return <h1>Loading...</h1>;
+  }
+
+  if (!data) {
+    return null;
+  }
+
+  async function handleSubmit(event) {
```

Comment:
The `handleSubmit` function currently only _**adds**_ new entries. It is not capable of _**updating**_ existing entries. That's because you only have API endpoints for GET, POST and DELETE but not for PUT.

Even if you select a Praline and change things around you will create _a new entry_ in the database instead of update the selected Praline. If you want to be able to update a Praline you need to add the case for PUT and call its endpoint.

Reply:
Update is part of my next userstory. 

All other suggestions I fixed except für deleting the usestate package. If I do so I get an error.

Reply:
I'd check where in your code you're using the package, because chances are you are using it accidentally rather than intentionally, seeing as how `useState` is built into React and doesn't need to be installed separately.